applications/nrf_desktop/src/modules/ble_discovery.c

```diff
@@ -267,8 +334,8 @@ static bool event_handler(const struct event_header *eh)
 			cast_module_state_event(eh);
 
 		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
-			k_work_init(&hids_discovery_start,
-				    hids_discovery_start_fn);
+			k_work_init(&next_discovery_step,
```

Comment:
Exctract code and create init function. Also initialize there state global variable (I know by default is has DISCOVERY_START value, but I prefere to have it explicitly initialized).

Reply:
done